src/GafferSceneUI/LightPositionTool.cpp

```diff
@@ -969,22 +973,26 @@ void LightPositionTool::setShadowTarget( const V3f &p, ScriptNodePtr scriptNode
 {
 	std::optional<V3f> currentValue = getShadowTarget();
 	auto shadowHandle = static_cast<ShadowHandle *>( m_shadowHandle.get() );
+	const auto pathString = ScenePlug::pathToString( selection().back().path() );
 	Action::enact(
 		scriptNode,
-		[this, p, shadowHandle]() {
-			m_shadowTargetMap[m_key] = p;
+		[t = LightPositionToolPtr( this ), k = pathString, p, shadowHandle]() {
+			t->m_shadowTargetMap[k] = p;
 			shadowHandle->setShadowTarget( p );
 		},
-		[this, currentValue, shadowHandle]() {
-			m_shadowTargetMap[m_key] = currentValue;
-			shadowHandle->setShadowTarget( currentValue );
+		[t = LightPositionToolPtr( this ), k = pathString, currentValue, shadowHandle]() {
+			t->m_shadowTargetMap[k] = currentValue;
+			if( ScenePlug::pathToString( t->selection().back().path() ) == k )
+			{
+				shadowHandle->setShadowTarget( currentValue );
+			}
 		}
 	);
 }
 
 std::optional<V3f> LightPositionTool::getShadowTarget() const
 {
-	auto it = m_shadowTargetMap.find( m_key );
+	auto it = m_shadowTargetMap.find( ScenePlug::pathToString( selection().back().path() ) );
```

Comment:
Again, `selection().back()` is being called here without checking that the selection is non-empty, and we're doing the same thing in a bunch of other places in this file. It's _possible_ that the check is being performed by the callers in all cases, but I haven't verified - in any case it seems fragile. I wonder if we should just use a little `selectedPathString()` utility function that returns `""` when there's no selection?

Reply:
I added the helper function in https://github.com/GafferHQ/gaffer/pull/5558/commits/53faf38062e7d976b24f39beb424bc63f7541f8b. I also double checked that the remaining `selection().back()` calls are protected behind a check for multiple items in `selection()`. All are, via `TransformTool::selectionEditable()` or checking that `selection().size() == 1`.